modules/sfp_countryname.py

```diff
@@ -0,0 +1,308 @@
+            "SL" : "Sierra Leone", "SG" : "Singapore", "SX" : "Sint Maarten",
+            "SK" : "Slovakia", "SI" : "Slovenia", "SB" : "Solomon Islands",
+            "SO" : "Somalia", "ZA" : "South Africa", "GS" : "South Georgia and the South Sandwich Islands",
+            "KR" : "South Korea", "SS" : "South Sudan", "ES" : "Spain",
+            "LK" : "Sri Lanka", "SD" : "Sudan", "SR" : "Suriname",
+            "SJ" : "Svalbard and Jan Mayen", "SZ" : "Swaziland", "SE" : "Sweden",
+            "CH" : "Switzerland", "SY" : "Syria", "TW" : "Taiwan",
+            "TJ" : "Tajikistan", "TZ" : "Tanzania", "TH" : "Thailand",
+            "TG" : "Togo", "TK" : "Tokelau", "TO" : "Tonga",
+            "TT" : "Trinidad and Tobago", "TN" : "Tunisia", "TR" : "Turkey",
+            "TM" : "Turkmenistan", "TC" : "Turks and Caicos Islands", "TV" : "Tuvalu",
+            "VI" : "U.S. Virgin Islands", "UG" : "Uganda", "UA" : "Ukraine",
+            "AE" : "United Arab Emirates", "GB" : "United Kingdom", "US" : "United States",
+            "UM" : "United States Minor Outlying Islands", "UY" : "Uruguay", "UZ" : "Uzbekistan",
+            "VU" : "Vanuatu", "VA" : "Vatican", "VE" : "Venezuela",
+            "VN" : "Vietnam", "WF" : "Wallis and Futuna", "EH" : "Western Sahara",
+            "YE" : "Yemen", "ZM" : "Zambia", "ZW" : "Zimbabwe", 
+            # Below are not country codes but recognized as TLDs
+            "AC" : "Ascension Island", "EU" : "European Union", "SU" : "Soviet Union", 
+            "UK" : "United Kingdom"
+        }
+        return abbvCountryCodes
+    
+    # Getter method for dictionary of codes not associated with any country
+    def getNonCountryCodesDict(self):
+        
+        # List of TLD not associated with any country
+        nonCountryCodes = ["COM", "NET", "ORG", "GOV", "MIL"]
+
+        # Get default country code set from opts dictionary
+        defaultCountryCode = self.opts["noncountrytlddefault"]
+
+        nonCountryCodesDict = dict()
+
+        # Set default country name for all keys
+        for nonCountryCode in nonCountryCodes: 
+            nonCountryCodesDict[nonCountryCode] = defaultCountryCode
+        
+        return nonCountryCodesDict
+    
+    # Detect name of country from phone number 
+    def detectCountryFromPhone(self, srcPhoneNumber):
+
+        # Get dictionary of country codes and country names
+        abbvCountryCodes = self.getCountryCodeDict()
+
+        # Parse source phone nummber
+        phoneNumber = phonenumbers.parse(srcPhoneNumber)
+        # Check if country code is present in the phone number
+        try:
+            countryCode = region_code_for_country_code(phoneNumber.country_code)
+        except:
+            # Region code not present in source phone number
+            self.debug("Skipped invalid phone number: " + srcPhoneNumber)
+            return None
+        return abbvCountryCodes[countryCode.upper()]
```

Comment:
If there is even a remote chance that the `countryCode` here doesn't exist in the dictionary above, best to use `abbvCountryCodes.get(countryCode.upper())`, otherwise this will raise an exception when the key doesn't exist. Using `.get()` will return None.